cartridges/int_algolia/cartridge/scripts/algolia/filters/productFilter.js

```diff
@@ -15,4 +15,25 @@ function isInclude(product) {
     return true;
 }
 
+/**
+ * If IndexOutOfStock is false, we only return `true` here if the product has
+ * ATS >= InStockThreshold. If IndexOutOfStock is true, then always return `true`.
```

Comment:
That's very complicated. Again, you must try to come with functions scoped and easy to understand and maintain.

Also, it's always better to write pure functions, i.e. that doesn't change behaviour due to external factor (here, the preferences)

To me the good thing to do here is:
- Have a simple  `isInStock(product, inStockThreshold)` function that tells you if the product is in stock
- Do this check of in the jobs:
  ```js
  if (!productFilter.isInStock(variant, ALGOLIA_IN_STOCK_THRESHOLD) && !INDEX_OUT_OF_STOCK) {
    continue;
  }
  ```